### DISCUSSION TITLE
Best Choice for File Operations in C: FILE-objects  or syscalls?

### DISCUSSION THREAD

**Discussion by mkaddani** [pyccel/pyccel#1467] [Polls]
In the context of issue https://github.com/pyccel/pyccel/issues/1351 text file operations in C for the Pyccel, we are faced with the decision of choosing between two methods: using (FILE-Objects) Fopen, fwrite, fclose functions, or utilizing (syscalls) open, write, close functions  for file manipulation. This discussion aims to explore the pros and cons of each approach and reach a consensus on the best method to use.

## Links
[Difference between system call and library call](https://www.geeksforgeeks.org/difference-between-system-call-and-library-call/)

**Comment by EmilyBourne** [comment]
> A code like this:
> ```python
> if __name__ == "__main__":
>     f = open('output.txt', 'wt', encoding='utf-8')
>     f.write('line')
>     f.close()
> ```
> 
> could be translated to c using syscalls:
> ```c
> int main()
> {
>     int fd = open("ouput.txt", O_WDONLY | O_CREAT);
>     write(fd, "line", 5)
>     close(fd);
> }
> ```
> 
> or using Fopen:
> ```c
> int main () {
>    FILE *fp;
>    fp = fopen( "output.txt" , "w" );
>    fwrite("line" , 1 , 5 , fp );
> 
>    fclose(fp);
>   
>    return(0);
> }
> ```

**Comment by EmilyBourne** [comment]
For me the winning argument is:
> A system call is not portable. | A library call is portable.

**Reply by EmilyBourne** [reply]
cc: @yguclu @saidctb @bauom

**Reply by bauom** [reply]
I agree with using FILE-objects, portability aside we have access to more function to operate on the file stream.
for example: `fprintf` `fseek` `rewind` ...
